### ISSUE TITLE
CuBERT Pretraining Dedup Process code? Identical SHAs in manifest

### ISSUE THREAD

**Issue by DNGros** [google-research/google-research#581]
When downloading the CuBERT pretraining corpus, I encountered some questions with the deduping process. Section 3.2 of [the paper](https://arxiv.org/pdf/2001.00059.pdf) mentions that there was a deduping process:

> .... This brought the dataset to 14.3 million files. We then further deduplicated the remaining files, by clustering them into equivalence classes holding similar files according to the same similarity metric, and keeping only one exemplar per equivalence class.This helps avoid biasing the pre-trained embedding. Finally,we removed files that could not be parsed. In the end, we were left with 7.4 million Python files containing over 9.3 billion tokens. This is our Python pre-training code corpus.

I downloaded [the manifest files](https://console.cloud.google.com/storage/browser/cubert/20200621_Python/github_python_minus_ethpy150open_deduplicated_manifest). There are indeed ~7.4M blobs. However, only about 4.1M of the SHAs are unique, implying over 3M point to identical text. There about 0.9M instances of the SHA e69de29bb2d1d6434b8b29ae775ad8c2e48c5391 which appears to correspond to an empty file. There are also thousands that correspond things like Django default files or 2626 instances of a SHA that corresponds to [a file from chardetection](https://github.com/splunk/TA-GitHubStats/blob/ce4eab957e6d9840d4230209b17d85d0162bc3e7/bin/chardet/euckrfreq.py) which could slightly confuse modeling. See [this gist](https://gist.github.com/DNGros/1a1bfaaac0544b1b53bf20a301d61b50) if want to see how I checked the SHAs.

Any thoughts on why there might be identical SHAs? I don't see the scripts used for deduping in the repo, but deduping is rather difficult, so I get there are ways it could go wrong in confusing ways.

CC: @adityakanade @maniatis

**Comment by maniatis** [comment]
Thank you so much for bringing this to our attention. I am updating the README to describe the reason for this duplication, and verified that it doesn't affect the separation between fine-tuning and pre-training datasets.
